src/cargo/core/compiler/mod.rs

```diff
@@ -672,6 +709,15 @@ fn add_cap_lints(bcx: &BuildContext, unit: &Unit, cmd: &mut ProcessBuilder) {
     }
 }
 
+fn add_color(bcx: &BuildContext, cmd: &mut ProcessBuilder) {
+    let capture_output = bcx.config.cli_unstable().compile_progress;
+    let shell = bcx.config.shell();
+    if capture_output || shell.color_choice() != ColorChoice::CargoAuto {
+        let color = if shell.supports_color() { "always" } else { "never" };
+        cmd.args(&["--color", color]);
```

Comment:
This is slightly different than the previous invocation that was moved up from below, but I think it's equivalent, right?

Previously if I pass `--color never` to Cargo it'll forward that to rustc unconditionally. I think that'll happen here because `stream.supports_color()` is already set to return `false` unconditionally inside of `termcolor`, I think?

(and I believe the reverse is true for `--color always`)

Just wanted to make sure!